pyext/src/macros.py

```diff
@@ -410,64 +426,66 @@ def execute_macro(self):
             self.rmf_output_objects.append(sw)
 
-        print("Setting up stat file")
-        output = IMP.pmi.output.Output(atomistic=self.vars["atomistic"])
-        low_temp_stat_file = globaldir + \
-            self.vars["stat_file_name_suffix"] + "." + str(myindex) + ".out"
+        if not self.nest:
+            print("Setting up stat file")
+            output = IMP.pmi.output.Output(atomistic=self.vars["atomistic"])
+            low_temp_stat_file = globaldir + \
+                self.vars["stat_file_name_suffix"] + "." + str(myindex) + ".out"
 
         # Ensure model is updated before saving init files
         if not self.test_mode:
             self.model.update()
 
-        if not self.test_mode:
+        if not self.test_mode and not self.nest:
             if self.output_objects is not None:
                 output.init_stat2(low_temp_stat_file,
                                   self.output_objects,
                                   extralabels=["rmf_file", "rmf_frame_index"])
         else:
             print("Stat file writing is disabled")
 
-        if self.rmf_output_objects is not None:
+        if self.rmf_output_objects is not None and not self.nest:
             print("Stat info being written in the rmf file")
 
-        print("Setting up replica stat file")
-        replica_stat_file = globaldir + \
-            self.vars["replica_stat_file_suffix"] + "." + str(myindex) + ".out"
-        if not self.test_mode:
-            output.init_stat2(replica_stat_file, [rex], extralabels=["score"])
+        if not self.test_mode and not self.nest:
+            print("Setting up replica stat file")
+            replica_stat_file = globaldir + \
+                self.vars["replica_stat_file_suffix"] + "." + str(myindex) + ".out"
+            if not self.test_mode:
+                output.init_stat2(replica_stat_file, [rex], extralabels=["score"])
 
-        if not self.test_mode:
-            print("Setting up best pdb files")
-            if not self.is_multi_state:
-                if self.vars["number_of_best_scoring_models"] > 0:
-                    output.init_pdb_best_scoring(
-                        pdb_dir + "/" + self.vars["best_pdb_name_suffix"],
-                        self.root_hier,
-                        self.vars["number_of_best_scoring_models"],
-                        replica_exchange=True,
-                        mmcif=self.vars['mmcif'],
-                        best_score_file=globaldir + "best.scores.rex.py")
-                    pdbext = ".0.cif" if self.vars['mmcif'] else ".0.pdb"
-                    output.write_psf(
-                        pdb_dir + "/" + "model.psf",
-                        pdb_dir + "/" +
-                        self.vars["best_pdb_name_suffix"] + pdbext)
-            else:
-                if self.vars["number_of_best_scoring_models"] > 0:
-                    for n in range(self.vars["number_of_states"]):
+            if not self.test_mode:
```

Comment:
This will always be True because it is inside a `if not self.test_mode and not self.nest:` block. Remove it and unindent the contained block pertaining to output best-scoring PDBs. That will also give a much cleaner diff.

Reply:
Done. Also, there is another similar `if not self.test_mode` on line 454 which can be removed as well.